docs/helpers.md

```diff
@@ -0,0 +1,97 @@
+Helper Structs
+--------------
+
+##### Memo
+
+Linked list, optimized for quickly finding a given type.
+
+- type: int, either a token or a rule (rules start at 256)
+- node: NULL or pointer to AST node OR pointer to token object
+- mark: if node != NULL, index into Parser's array of tokens
+- next: NULL or pointer to next Memo structure
+
+##### Token
+
+These are in an array linked from Parser.  (Or linked list???)
```

Comment:
It's an array. We use a linked list for `Memo` only. (Maybe we should explain how the data structures fit together? It would be best explained using a diagram.)

Reply:
I don't really have a clear enough picture of how this should look. Should we maybe merge this PR without the diagram and do that later?